funcx_sdk/funcx/sdk/web_client.py

```diff
@@ -135,6 +139,30 @@ def get_batch_status(
             data.update(additional_fields)
         return self.post("/batch_status", data=data)
 
+    # the FuncXClient needs to send version information through BaseClient.app_name,
+    # so that's overridden here to prevent direct manipulation. use user_app_name
+    # instead to send any custom metadata through the User Agent request header
+    @property
+    def app_name(self) -> t.Optional[str]:
+        return super().app_name
+
+    @app_name.setter
+    def app_name(self, value) -> None:
+        raise NotImplementedError("Use user_app_name instead")
```

Comment:
Manipulating properties of a superclass is always messy. I think what you've done here is about as "clean" as it can get. You define a new property and then give it a setter.

I'd be okay with merging this, but before we do I want to ask: do we need to modify the `app_name` property at all?

What about
```python
@property
def user_app_name(self) -> str:
    return self.app_name

@user_app_name.setter
def user_app_name(self, value: str) -> None:
    app_name = f"funcx-sdk-{__version__}"
    if value is not None:
        app_name += f"/{value}"
    self.app_name = app_name
```
?

It seems to me like there are quite a few steps here, in service of making sure that the caller can't set `app_name`. If we give up on that constraint, then I think all of this simplifies quite a bit.

Reply:
I like that approach for the simplicity, but it does allow users to override our version information if they ever set `app_name` directly. I'm admittedly not sure how much of an impact that would have on debugging, since we can always at least ask users to run `pip freeze`, but I feel like it goes against the spirit of the story to deprive ourselves of logging that information in any circumstance.

Reply:
I feel like it's fine to allow a user to squash/discard this info since it's purely a "courtesy" from the client code to the service (User-Agent is notoriously full of lies), but I'm also not strongly _against_ taking this protective measure. So let's approve/merge with the current code! We can always adjust later if we want.